packages/augur-sdk/src/state/db/LiquidityDB.ts

```diff
@@ -39,50 +51,47 @@ export class LiquidityDB extends AbstractDB {
     });
   }
 
-  async getMarketsLiquidity(marketIds?: string[]): Promise<any[]> {
-    const currentTimestamp = new BigNumber(Math.floor(Date.now() / 1000));
-    const secondsPerHour = SECONDS_IN_AN_HOUR.toNumber();
-    const mostRecentOnTheHourTimestamp = currentTimestamp.minus(currentTimestamp.mod(secondsPerHour));
-    const selectorConditions: any[] = [
-      { _id: { $ne: 'lastUpdated' } },
-      { timestamp: { $gte: mostRecentOnTheHourTimestamp.minus(SECONDS_IN_A_DAY).toNumber() } },
-    ];
-    if (marketIds) {
-      selectorConditions.push(
-        { market: { $in: marketIds } }
-      );
-    }
-    const marketsLiquidity = await this.db.find({
-      selector: {
-        $and: selectorConditions,
-      },
-    });
-    return marketsLiquidity.docs;
-  }
-
-  async recalculateLiquidity(augur: Augur, db: DB, timestamp: number): Promise<void> {
+  async recalculateLiquidity(augur: Augur, db: DB, currentTimestamp: number): Promise<void> {
+  console.log('currentTimestamp', currentTimestamp);
     // @TODO: Need to factor in blockStreamDelay?
     try {
       const liquidityDB = db.getLiquidityDatabase();
-      const lastUpdated: any = await liquidityDB.getDocument('lastUpdated');
-      const currentTimestamp = new BigNumber(timestamp);
+      const lastUpdatedTimestamp = await db.findLiquidityLastUpdatedTimestamp();
+      const currentTimestampBN = new BigNumber(currentTimestamp);
       const secondsPerHour = SECONDS_IN_AN_HOUR.toNumber();
-      const mostRecentOnTheHourTimestamp = currentTimestamp.minus(currentTimestamp.mod(secondsPerHour));
-      if (!lastUpdated || mostRecentOnTheHourTimestamp.gt(lastUpdated.timestamp)) {
+      const mostRecentOnTheHourTimestamp = currentTimestampBN.minus(currentTimestampBN.mod(secondsPerHour));
```

Comment:
I'm not sure about this logic. Is `mostRecentOnTheHourTimestamp` suppose to be the next time (future) to calculate liquidity?

Reply:
It's the most recent on-the-hour timestamp, meaning if the current time is 2:45pm, the most recent on-the-hour timestamp is 2:00pm.